### PR TITLE
HIP and Spack

### PR DESCRIPTION
**Pull request by sekelle**
* Compile HIP directly without hipification
* Updated CMake config to simplify spack package implementation
* Organize N-body code into more translation units and remove duplicate upsweep and mass-center implementations

### CODE REVIEW

File: domain/include/cstone/traversal/groups_gpu.cu [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        computeGroupSplitsImpl<GpuConfig::warpSize>(first, last, x, y, z, h, leaves, numLeaves, layout, box, tolFactor,
+                                                    splitMasks, numSplitsPerGroup, groups);
+    }
+    else if (groupSize == 2 * GpuConfig::warpSize)
+    {
+        DeviceVector<util::array<GpuConfig::ThreadMask, 2>> splitMasks;
+        computeGroupSplitsImpl<2 * GpuConfig::warpSize>(first, last, x, y, z, h, leaves, numLeaves, layout, box,
+                                                        tolFactor, splitMasks, numSplitsPerGroup, groups);
+    }
+    else { throw std::runtime_error("Unsupported spatial group size\n"); }
```
Comment (jgphpc) [inline domain/include/cstone/traversal/groups_gpu.cu:148]: on x-daint, I get:
```
groups_gpu.cu(146): error: 
namespace "std" has no member "runtime_error"
```
with gcc/11.2, cuda/11.6 (stdexcept)

Comment (sekelle) [reply domain/include/cstone/traversal/groups_gpu.cu:148]: should be fixed now

**Comment by jgphpc** [comment]
cscs-ci run

File: README.md [hunk 2]
```diff
@@ -109,7 +109,7 @@ Module and CMake configuration on LUMI
 ```shell
 module load CrayEnv buildtools/22.12 craype-accel-amd-gfx90a rocm cray-hdf5-parallel
 cd <GIT_SOURCE_DIR>; hipify-perl -inplace `find -name *.cu -o -name *.cuh` && find -name *.prehip -delete
```
Comment (jgphpc) [inline README.md:111]: hipify still needed ?

Comment (sekelle) [reply README.md:111]: No, let me remove this

**Review by jgphpc — APPROVED** [review]

**Pull request merged**